File: medium/151. Reverse Words in a String/151. Reverse Words in a String.ts
function reverseWords(s: string): string {
    if (!s.length)
        return ""
    // let a = s.trim().split(" ").filter(Boolean)
    // return a.reverse().join(" ")

    s = s.trim();
    return s.split(/\s+/).reverse().join(' ');
};


const s_1 = "the sky is blue"
const s_2 = "  hello world  "
const s_3 = "a good   example"

console.log(reverseWords(s_1))
console.log(reverseWords(s_2))
console.log(reverseWords(s_3))

File: medium/151. Reverse Words in a String/151. Reverse Words in a String.py
class Solution:
    def reverseWords(self, s: str) -> str:
        # if len(s) == 1:
        #     return s
        # s = s.strip()
        # i = 0
        # while i < len(s):
        #     if s[i] == " " and s[i + 1] == " ":
        #         s = s[:i] + s[i + 1 :]
        #     else:
        #         i += 1
        # s_arr = s.split(" ")
        # s_arr.reverse()
        # return " ".join(s_arr)

        a = s.strip().split()
        return "".join(a[::-1])


solution = Solution()

s_1 = "the sky is blue"
s_2 = "  hello world  "
s_3 = "a good   example"

print(solution.reverseWords(s_1))
print(solution.reverseWords(s_2))
print(solution.reverseWords(s_3))
